### PR TITLE
Add csv export to scripting api

### PR DESCRIPTION
**Pull request by todvora**
Added support for CSV export

## How Has This Been Tested?
Added integration tests

## Screenshots (if appropriate):
image

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java [hunk 1]
```diff
@@ -267,6 +271,71 @@ void testGetRequestAcii() {
...
+                .statusCode(200)
+                .extract().body().asInputStream();
+
+        final CsvParser csvParser = new CsvParser(Csv.parseRfc4180());
+        final List<String[]> lines = csvParser.parseAll(response);
+
+
+
+        // headers
+        Assertions.assertArrayEquals(lines.get(0), new String[]{"Grouping", "Metric : count"});
```
Comment (luk-kaminski) [inline full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java:307]: I'd consider replacing "Metric : count" with ResponseSchemaEntry.metric(...).name()

**Review by luk-kaminski — APPROVED** [review]
Looks good, thank you!

**Review by dennisoelkers — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java [hunk 2]
```diff
@@ -141,9 +145,34 @@ public String executeQueryAsciiOutput(@ApiParam(name = "searchRequestSpec") @Val
         return at.render();
     }
 
+    @POST
+    @ApiOperation(value = "Execute aggregation specified by `searchRequestSpec`")
+    @Path("aggregate")
+    @NoAuditEvent("Creating audit event manually in method body.")
+    @Produces(MoreMediaTypes.TEXT_CSV)
+    public Response executeQueryCsvOutput(@ApiParam(name = "searchRequestSpec") @Valid SearchRequestSpec searchRequestSpec,
```
[the remark this answers is not in the document]
Comment (todvora) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java]: Good point, implemented. Thanks!

File: full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java (hunk 1 above)
Comment (todvora) [reply full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java]: IDK, I feel that's better to have hardcoded values in the test. If the name() method fails and delivers empty string, we won't catch that in the test.

**Review by dennisoelkers — APPROVED** [review]

**Pull request merged**